code/game/objects/structures/mirror.dm

```diff
@@ -133,31 +135,33 @@
 			playsound(src, 'sound/effects/hit_on_shattered_glass.ogg', 70, TRUE)
 
 
+/**
+ * Magical mirrors, which let you change your name, gender, race and also your hair still. Vary in power.
+ */
 /obj/structure/mirror/magic
 	name = "magic mirror"
 	desc = "Turn and face the strange... face."
 	icon_state = "magic_mirror"
-	var/list/choosable_races = list()
-
-/obj/structure/mirror/magic/New()
-	if(!choosable_races.len)
-		for(var/speciestype in subtypesof(/datum/species))
-			var/datum/species/S = speciestype
-			if(initial(S.changesource_flags) & MIRROR_MAGIC)
-				choosable_races += initial(S.id)
-		choosable_races = sortList(choosable_races)
-	..()
-
-/obj/structure/mirror/magic/lesser/New()
-	choosable_races = GLOB.roundstart_races.Copy()
-	..()
+	var/list/choosable_races
 
-/obj/structure/mirror/magic/badmin/New()
+/obj/structure/mirror/magic/Initialize()
+	. = ..()
+	choosable_races = list()
+	setup_choosable_races()
+
+/**
+ * Determines what races a user of this magic mirror have access to.
+ *
+ * By default, uses the MIRROR_MAGIC flag on /datum/species, but
+ * subtypes can override this without calling parent to specify more
+ * or less restrictions.
+ */
+/obj/structure/mirror/magic/proc/setup_choosable_races()
```

Comment:
Maybe save this into a global, instead of making one list per mirror?

Reply:
So we're making a global list per mirror type?

Reply:
That or you can use type lists.
Given it's just a few lists and they have custom setup procs global lists work fine IMO, but up to you.